Microsoft.Toolkit.Uwp.UI.Animations/CompositionAnimations/Animations/AnimationBase.cs

```diff
@@ -0,0 +1,147 @@
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using Windows.Foundation.Metadata;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Markup;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// Abstract class providing common dependency properties for composition animations
+    /// </summary>
+    [ContentProperty(Name = nameof(KeyFrames))]
+    public abstract class AnimationBase : DependencyObject
+    {
+        /// <summary>
+        /// Identifies the <see cref="Target"/> property
+        /// </summary>
+        public static readonly DependencyProperty TargetProperty =
+            DependencyProperty.Register("Target", typeof(string), typeof(AnimationBase), new PropertyMetadata(string.Empty, OnAnimationPropertyChanged));
+
+        /// <summary>
+        /// Identifies the <see cref="Duration"/> property
+        /// </summary>
+        public static readonly DependencyProperty DurationProperty =
+            DependencyProperty.Register("Duration", typeof(TimeSpan), typeof(AnimationBase), new PropertyMetadata(TimeSpan.FromMilliseconds(400), OnAnimationPropertyChanged));
+
+        /// <summary>
+        /// Identifies the <see cref="KeyFrames"/> property
+        /// </summary>
+        public static readonly DependencyProperty KeyFramesProperty =
+            DependencyProperty.Register("KeyFrames", typeof(KeyFrameCollection), typeof(AnimationBase), new PropertyMetadata(null, OnAnimationPropertyChanged));
+
+        /// <summary>
+        /// Identifies the <see cref="ImplicitTarget"/> property
+        /// </summary>
+        public static readonly DependencyProperty ImplicitTargetProperty =
+            DependencyProperty.Register("ImplicitTarget", typeof(string), typeof(AnimationBase), new PropertyMetadata(null, OnAnimationPropertyChanged));
+
+        /// <summary>
+        /// Identifies the <see cref="Delay"/> property
+        /// </summary>
+        public static readonly DependencyProperty DelayProperty =
+            DependencyProperty.Register("Delay", typeof(TimeSpan), typeof(AnimationBase), new PropertyMetadata(TimeSpan.Zero, OnAnimationPropertyChanged));
+
+        /// <summary>
+        /// Raised when a property changes
+        /// </summary>
+        public event EventHandler AnimationChanged;
+
+        /// <summary>
+        /// Gets or sets the duration of the animation
+        /// </summary>
+        public TimeSpan Duration
+        {
+            get { return (TimeSpan)GetValue(DurationProperty); }
+            set { SetValue(DurationProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="KeyFrameCollection"/> of the animations
+        /// </summary>
+        public KeyFrameCollection KeyFrames
+        {
+            get
+            {
+                var collection = (KeyFrameCollection)GetValue(KeyFramesProperty);
```

Comment:
You should never have code in get/setters of dependency properties (there's no guarantee this is called if you're using GetValue). Instead initialize in the constructor

Reply:
This removes the need to specify a KeyFrameCollection in XAML. For example, with this, I can add the keyframes like so:
``` 
<Animation>
    <KeyFrame />
    <KeyFrame />
</Animation>
```
Otherwise, I would have to specify the collection:
``` 
<Animation>
    <KeyFrameCollection>
        <KeyFrame />
        <KeyFrame />
    </KeyFrameCollection>
</Animation>
```
I could initialize a new KeyFrameCollection in the constructor, but why have an instance if it's not needed? Open to other suggestions?